File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
      queue<TreeNode*> q;
        vector<vector<int>> ans;
        vector<int> sol;
        q.push(root);
        q.push(NULL);
        while(!q.empty() && q.front()!=NULL )
        {
            TreeNode* front= q.front();
            q.pop();
            
            if(front->left)
                q.push(front->left);
            if(front->right)
                q.push(front->right);
            sol.push_back(front->val);
            if(q.front()==NULL)
            {
                q.pop();
                q.push(NULL);
                ans.push_back(sol);
                sol.clear();
                
            }    
            
        }  
        return ans;

    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        ans = []
        if not root:
            return []
        q = deque()
        q.append(root)
        while q:
            l = len(q)
            res = []
            for i in range(l):
                cur  = q.popleft()
                if cur:
                    res.append(cur.val)
                    if cur.left:
                        q.append(cur.left)
                    if cur.right:    
                        q.append(cur.right)        
            ans.append(res)
        return ans   


        
             

                    
                      


                
                
                
                
                


        